### PR TITLE
Add compatibility signup

### PR DESCRIPTION
**Pull request by eliegaboriau**
This PR slightly change the fonctionnement of the module to make it compatible with the friendly sigup module.

It now asks for the captcha validation on a specific context, so that it is not checked when using the instant validation feature.

### CODE REVIEW

File: lib/decidim/question_captcha/engine.rb [hunk 1]
```diff
@@ -32,6 +32,12 @@ class Engine < ::Rails::Engine
           Decidim::RegistrationForm.class_eval do
             extend(ActsAsTextcaptcha::Textcaptcha)
             include(Decidim::QuestionCaptcha::HasCaptcha)
+
+            validate :validate_textcaptcha, if: :perform_textcaptcha?, on: :validate_captcha
```
Comment (armandfardeau) [inline lib/decidim/question_captcha/engine.rb:36]: Could you move this line to `Decidim::QuestionCaptcha::HasCaptcha`?

**Review by armandfardeau — CHANGES_REQUESTED** [review]

File: lib/decidim/question_captcha/engine.rb (hunk 1 above)
Comment (eliegaboriau) [reply lib/decidim/question_captcha/engine.rb]: It should be ok now @armandfardeau

File: Gemfile.lock [hunk 2]
```diff
@@ -781,7 +781,7 @@ DEPENDENCIES
   web-console (~> 3.5)
 
 RUBY VERSION
-   ruby 2.7.5p203
+   ruby 2.7.1p83
```
Comment (Quentinchampenois) [inline Gemfile.lock:784]: Ruby version in `Gemfile.lock` must always be equal to [.ruby-version](https://github.com/OpenSourcePolitics/decidim-module-question_captcha/blob/add_compatibility_signup/.ruby-version) 

Can you revert this change pls ?

**Review by Quentinchampenois — CHANGES_REQUESTED** [review]
Just a change, also can you add a description to explain context please ?

**Review by armandfardeau — APPROVED** [review]

**Review by armandfardeau — APPROVED** [review]

**Review by paulinebessoles — APPROVED** [review]
